### DISCUSSION TITLE
win7下的vscode终端字符颜色显示乱码

### DISCUSSION THREAD

**Discussion by zyjdmmm** [xmake-io/xmake#3591] [Q&A]
### Xmake 版本

2.7.7

### 操作系统版本和架构

win 7企业版   vscode1.70.3

### 描述问题

由于一些原因我不得不使用win7，我找到了最后一个支持win7的vscode版本1.70.3。
但是在vscode的poershell终端似乎xmake颜色字符乱码了，然后从启动直接使用powershell又没有乱码。
![vscode中的powershell终端显示乱码](https://user-images.githubusercontent.com/75295100/228297454-4eea8b19-d31f-405d-b988-0819283d29e1.PNG)

![我直接使用powershell又没有乱码](https://user-images.githubusercontent.com/75295100/228520253-1a8cd7b1-d7a6-40db-a6bc-a5c31e0576fb.PNG)

在win7下关闭xmake的颜色显示能否有用？

### 期待的结果

nothing

**Comment by zyjdmmm** [comment]
win7下`xmake g --theme=plain`关闭颜色有用，但能否自动一点点

**Comment by waruqi** [comment]
原本 win7 下就是禁用的，可以看下是否设置了 $ANSICON 

也可以调下 https://github.com/xmake-io/xmake/blob/884ec302caa4751a18a5a85de04e64a42ac8b210/xmake/core/base/tty.lua#L322

**Comment by zyjdmmm** [comment]
> 原本 win7 下就是禁用的，可以看下是否设置了 $ANSICON
> 
> 也可以调下
> 
> https://github.com/xmake-io/xmake/blob/884ec302caa4751a18a5a85de04e64a42ac8b210/xmake/core/base/tty.lua#L322

打印了一下似乎并没有这个环境变量：
![捕获](https://user-images.githubusercontent.com/75295100/228525025-892fdd40-aa5f-499b-a808-284204bd0055.PNG)
如前面所说颜色乱码只出现在vscode的终端中，直接打开powrshell终端字符颜色是关掉的没有出现乱码。

**Comment by waruqi** [comment]
应该是 vscode 下终端带了一些环境变量。比如 COLORTERM 都会影响 color 输出，这种暂时无解，除非它不设置，否则会干扰 探测，如果它设置了 COLORTERM，理论上也是 vscode 自身对 colorterm 探测有问题

**Comment by wzhy** [comment]
我在windows 7下也遇到了一模一样的问题，不知道楼主解决了吗？
